training/components.py

```diff
@@ -292,154 +230,175 @@ def list_phase1_final_model():
-                      image: {image}
-                      name: pytorch
-                      volumeMounts:
-                        - mountPath: /input_data
-                          name: input-data
-                          readOnly: true
-                        - mountPath: /input_model
-                          name: model
-                          readOnly: true
-                        - mountPath: /output
-                          name: output
-                          readOnly: true
-                      env:
-                        - name: NNODES
-                          value: \"{nnodes}\"
-                        - name: NPROC_PER_NODE
-                          value: \"{nproc_per_node}\"
-                        - name: XDG_CACHE_HOME
-                          value: /tmp
-                        - name: TRITON_CACHE_DIR
-                          value: /tmp
-                        - name: HF_HOME
-                          value: /tmp
-                        - name: TRANSFORMERS_CACHE
-                          value: /tmp
-                      resources:
-                        requests:
-                          "nvidia.com/gpu": {nproc_per_node}
-                        limits:
-                          "nvidia.com/gpu": {nproc_per_node}
-                  volumes:
-                    - name: input-data
-                      persistentVolumeClaim:
-                        claimName: {input_pvc_name}
-                    - name: model
-                      persistentVolumeClaim:
-                        claimName: {model_pvc_name}
-                    - name: output
-                      persistentVolumeClaim:
-                        claimName: {output_pvc_name}
-        """
+            """
+    ]
+
+    # Set volumes
+    volumes = [
+        models.V1Volume(
+            name="input-data",
+            persistent_volume_claim=models.V1PersistentVolumeClaimVolumeSource(
+                claim_name=input_pvc_name
+            ),
+        ),
+        models.V1Volume(
+            name="model",
+            persistent_volume_claim=models.V1PersistentVolumeClaimVolumeSource(
+                claim_name=model_pvc_name
+            ),
+        ),
+        models.V1Volume(
+            name="output",
+            persistent_volume_claim=models.V1PersistentVolumeClaimVolumeSource(
+                claim_name=output_pvc_name
+            ),
+        ),
+    ]
+
+    # Set volume mounts
+    volume_mounts_common = [
+        models.V1VolumeMount(
+            mount_path="/input_data", name="input-data", read_only=True
+        ),
+        models.V1VolumeMount(mount_path="/input_model", name="model", read_only=True),
+    ]
+    volume_mounts_master = volume_mounts_common.append(
+        models.V1VolumeMount(mount_path="/output", name="output")
+    )
+    volume_mounts_worker = volume_mounts_common.append(
+        models.V1VolumeMount(mount_path="/output", name="output", read_only=True)
```

Comment:
This doesn't work. `list.append()` returns `None` in Python and modifies the original array:


```python
>>> a = []
>>> b = a.append("foo")
>>> a
['foo']
>>> b
# Nothing is returned
```

Reply:
Fixed, thanks 👍